libautoscoper/src/PSO.cpp

```diff
@@ -46,81 +115,62 @@ float getRandomClamped()
   return (float)rand() / (float)RAND_MAX;
 }
 
-void pso(float *positions, float *velocities, float *pBests, float *gBest, unsigned int MAX_EPOCHS, unsigned int MAX_STALL)
+void pso(std::vector<Particle>& particles, Particle& gBest, unsigned int MAX_EPOCHS, unsigned int MAX_STALL)
 {
   int stall_iter = 0;
-  float tempParticle1[NUM_OF_DIMENSIONS];
-  float tempParticle2[NUM_OF_DIMENSIONS];
-
   bool do_this = true;
   unsigned int counter = 0;
-  //for (int iter = 0; iter < (signed int)MAX_EPOCHS; iter++)
-
   float OMEGA = 0.8f;
 
+  // Make a copy of the particles, this will be the initial pBest
+  std::vector<Particle> pBest = particles;
+
+  // Calc NCC for gBest
+  gBest.NCC = host_fitness_function(gBest.Position);
```

Comment:
```suggestion
```

I think we can remove this call. The loop below already takes care of computing it.